internal/bminventory/inventory_v2_handlers.go

```diff
@@ -671,6 +671,26 @@ func (b *bareMetalInventory) V2SetIgnoredValidations(ctx context.Context, params
 	return installer.NewV2SetIgnoredValidationsCreated().WithPayload(&ignoredValidations)
 }
 
+func (b *bareMetalInventory) V2GetClusterUISettings(ctx context.Context, params installer.V2GetClusterUISettingsParams) middleware.Responder {
+	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return installer.NewV2GetClusterUISettingsOK().WithPayload(cluster.UISettings)
+}
+
+func (b *bareMetalInventory) V2UpdateClusterUISettings(ctx context.Context, params installer.V2UpdateClusterUISettingsParams) middleware.Responder {
+	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	cluster.UISettings = params.UISettings
+	if err = b.db.Save(cluster).Error; err != nil {
```

Comment:
Generally I'd advise against this and say it's outright incorrect without a transaction.

This will re-save every field in `cluster` which will override any other requests that came in between when we got the cluster and when we saved it.

If the one line with `Updates` doesn't work then I'm fine with one call checking for existence then the second call being the update to only the one column.